src/app/shared/directives/editable-div.directive.ts

```diff
@@ -0,0 +1,48 @@
+import {Directive, forwardRef, ElementRef, HostListener, Renderer2} from '@angular/core';
+import {ControlValueAccessor, NG_VALUE_ACCESSOR} from '@angular/forms';
+
+@Directive({
+  selector: '[editable-div]',
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => EditableDivDirective),
+      multi: true
+    }
+  ]
+})
+export class EditableDivDirective implements ControlValueAccessor {
```

Comment:
Just a question: What problems does this directive solve?

Reply:
Reactive forms needs this for two-way-binding.

Reply:
This technique throws deprecation warning in Angular6. See: https://angular.io/api/forms/FormControlName#use-with-ngmodel

I tried to remove [(ngModel)] and directive but it throws error: `no value accessor attached....`. I'm confused now.

Reply:
AFAIK template driven forms use two-way data binding and reactive forms should be manipulated by form controls. But two-way data binding is not a good practice in general.

I think in this case you need to set `control.value` on `queryItem` input change and then react to `(input)` event on the editable `div` and update the query item.